tools/deploy-to-svn.sh

```diff
@@ -1,110 +1,156 @@
+	elif [[ $(ls -A -- "$BUILD_DIR") ]]; then
+		proceed_p "Directory $BUILD_DIR already exists, and is not empty." "Delete it?"
 	fi
+	rm -rf "$BUILD_DIR"
+	mkdir -p "$BUILD_DIR"
+fi
+cd "$BUILD_DIR"
+DIR=$(pwd)
+echo "Using build dir $DIR"
+
+echo "Checking mirror repo"
+git init -q -b tmp .
+git remote add origin "https://github.com/${MIRROR}.git"
+git remote update
+if [[ "$(git ls-remote --tags origin "$TAG" 2>/dev/null)" ]]; then
+	: # Tag exists
+elif [[ "$(git ls-remote --heads origin "$TAG" 2>/dev/null)" ]]; then
+	proceed_p "You are about to deploy a change from an unstable state 'HEAD'. This should only be done to update string typos for translators."
+else
+	echo "Tag $TAG not found in git repository. Please try again with a valid tag." >&2
+	exit 1
 fi
 
-git checkout $TARGET
-
-# Prep a home to drop our new files in. Just make it in /tmp so we can start fresh each time.
-rm -rf $JETPACK_SVN_DIR
-
-echo "Checking out SVN shallowly to $JETPACK_SVN_DIR"
-svn -q checkout https://plugins.svn.wordpress.org/jetpack/ --depth=empty $JETPACK_SVN_DIR
+echo "Checking out SVN shallowly to $DIR"
+svn -q checkout "https://plugins.svn.wordpress.org/$WPNAME/" --depth=empty "$DIR"
 echo "Done!"
 
-cd $JETPACK_SVN_DIR
-
-echo "Checking out SVN trunk to $JETPACK_SVN_DIR/trunk"
+echo "Checking out SVN trunk to $DIR/trunk"
 svn -q up trunk
 echo "Done!"
 
-echo "Checking out SVN tags shallowly to $JETPACK_SVN_DIR/tags"
+echo "Checking out SVN tags shallowly to $DIR/tags"
 svn -q up tags --depth=empty
 echo "Done!"
 
 echo "Deleting everything in trunk except for .svn directories"
-for file in $(find $JETPACK_SVN_DIR/trunk/* -not -path "*.svn*"); do
-	rm $file 2>/dev/null
-done
-echo "Done!"
-
-echo "Rsync'ing everything over from Git except for .git stuffs"
-rsync -r --exclude='*.git*' $JETPACK_GIT_DIR/* $JETPACK_SVN_DIR/trunk
+find trunk ! \( -path '*/.svn/*' -o -path "*/.svn" \) \( ! -type d -o -empty \) -delete
+[[ -e trunk ]] || mkdir -p trunk # If there were no .svn directories, trunk itself might have been removed.
 echo "Done!"
 
-echo "Purging paths included in .svnignore"
-# check .svnignore
-shopt -s globstar # Support globs.
-for file in $( cat "$JETPACK_GIT_DIR/.svnignore" 2>/dev/null ); do
-	rm -rf $JETPACK_SVN_DIR/trunk/$file
-done
+echo "Checking out $MIRROR $TAG into trunk"
+mv .git trunk/
+cd trunk
+git fetch --depth=1 origin "$TAG"
+git checkout -q FETCH_HEAD
 echo "Done!"
 
-echo "Remove empty directories"
-ANY=1
-while [[ -n "$ANY" ]]; do
-	ANY=
-	for empty_dir in $( find $JETPACK_SVN_DIR/trunk/* -type d -empty ); do
-		rm -rf $empty_dir
-		ANY=1
-	done
-done
+echo "Removing .git files and empty directories"
+find . -name '.git*' -print -exec rm -rf {} +
+find . -type d -empty -print -delete
 echo "Done!"
 
-echo "Generating Jetpack CDN Manifest"
-php ./trunk/bin/build-asset-cdn-json.php
-echo "Done!"
+echo "Your SVN checkout is at $DIR"
 
 # Tag the release.
-# svn cp trunk tags/$TARGET
```

Comment:
I would be okay in an interation with making these available via the script with a strong confirmation, e.g. "type 'Yes, I want to release this to SVN.` to proceed.

That said, it looks like we're lacking any steps to add/rm files that were added/removed so these commands are more of a suggestion than something ready to use as it.